src/background.js

```diff
@@ -56,25 +62,30 @@ chrome.storage.local.get(storedState => {
             if( state[item] !== changes[item].newValue){
                 state[item] = changes[item].newValue
                 if(item === "enabled"){
-                    state.enabled ? attachListeners() : detachListeners()
+                    state.enabled ? attachListeners(isHttps(state.proxyUrl)) : detachListeners()
                     chrome.browserAction.setIcon({
                         path: state.enabled ? 'assets/icon-128.png' : 'assets/icon-128-disabled.png'
                     })
                 }
+                if(item === "proxyUrl"){
+                    if(state.enabled){
```

Comment:
can these two `if`s be combined?

Reply:
I ended up changing it to be more of an object-literal case-switch implementation, just in case other properties need to be targeted with changeHandlers